### ISSUE TITLE
[BUG] `0.34.0` breaks OkHttp users on Gradle

### ISSUE THREAD

**Issue by boris-petrov** [langchain4j/langchain4j#1780]
**Describe the bug**
Using version 0.34.0 of the BOM and having a transitive OkHttp dependency on Gradle breaks. Please check from [this comment](https://github.com/langchain4j/langchain4j/pull/1645#issuecomment-2350934619) for a lot more details.

**Log and Stack trace**
See the reproduction in the link above.

**To Reproduce**
See the reproduction in the link above.

**Expected behavior**
Version 0.34.0 to work like 0.33.0.

**Please complete the following information:**
- LangChain4j version: 0.34.0
- LLM(s) used: N/A
- Java version: N/A
- Spring Boot version (if applicable): N/A

**Additional context**
cc @PrimosK

**Comment by langchain4j** [comment]
@boris-petrov could you please check the latest version (0.36.2 currently)? Does this problem still exist?

**Comment by boris-petrov** [comment]
Yes, the problem is the same. Please check [this comment](https://github.com/langchain4j/langchain4j/pull/1645#issuecomment-2350963832) for a reproduction repo and description on what to do. I've updated it to use the latest `langchain4j`.

**Comment by langchain4j** [comment]
@kpavlov could you please help with this? 🙏

**Comment by kpavlov** [comment]
@langchain4j , the explicit dependency to `kotlin-stdlib*` was already removed and `kotlin-bom `was added to `langchain4j-parent` to enforce dependency versions. I can only propose #2221 for further cleanup and version alignment.

In consumer projects, I can also recommend adding explicit dependency or use BOMs to resolve a dependency conflict. It's so many different combinations, so we can't predict them all.

Proactively dependency upgrade PRs by Renovate can help to prevent huge gaps between current and used versions. @langchain4j would you mind to spend some time reviewing renovate PRs, it's quite a long [list](https://github.com/langchain4j/langchain4j/pulls/app%2Frenovate) already. And the list  op pending (rate-limited) PRs is even longer: #2069
